components/eamxx/src/share/field/field_impl.hpp

```diff
@@ -693,6 +693,34 @@ scale (const ST beta)
   }
 }
 
+template<HostOrDevice HD>
+void Field::
+scale_inv (const Field& x)
+{
+  const auto& dt = data_type();
+  if (dt==DataType::IntType) {
+    int fill_val = constants::DefaultFillValue<int>().value;
+    if (get_header().has_extra_data("mask_value")) {
+      fill_val = get_header().get_extra_data<int>("mask_value");
+    }
+    return update_impl<CombineMode::Multiply,HD,int>(x,0,0,fill_val);
+  } else if (dt==DataType::FloatType) {
+    float fill_val = constants::DefaultFillValue<float>().value;
+    if (get_header().has_extra_data("mask_value")) {
+      fill_val = get_header().get_extra_data<float>("mask_value");
+    }
+    return update_impl<CombineMode::Multiply,HD,float>(x,0,0,fill_val);
```

Comment:
Was this and the above meant to have CombineMode::Divide, like the Double example below?

Reply:
Ah rats, I missed those. Thanks.